File: src/get-data.ts
import { SystemCommand } from "./systemCommands";
import { CommandType } from "./cmd";
import getType from "./get-type";

export default function getData(data: SystemCommand): null | Buffer {
    const type = getType(data);
    let out: Buffer | null = null;

    switch (type) {
        case CommandType.VimAfter:
            case CommandType.VimInsert:
            out = Buffer.from(`norm ${type === CommandType.VimAfter ? "a" : "i"}${data.message}`);
        break;

        case CommandType.VimCommand:
            out = Buffer.from(`norm ${data.message}`);
        break;

        case CommandType.SystemCommand:
            out = Buffer.from(`silent! !${data.message}`);
        break;
    }
    return out;
}

File: pkg/irc/getData.go
package irc

import "fmt"


func GetData(data IrcMessage) []byte {
    getType := GetType(data)
    var out []byte

    switch getType {
    case VimAfter:
        out = []byte(fmt.Sprintf("norm a%s", data.Message))
    case VimInsert:
        out = []byte(fmt.Sprintf("norm i%s", data.Message))
    case VimCommand:
        out = []byte(fmt.Sprintf("norm %s", data.Message))
    case SystemCommand:
        out = []byte(fmt.Sprintf("silent! !%s", data.Message))
    }

    return out
}
